lib/libxdp/xsk.c

```diff
@@ -696,6 +780,34 @@ static int __xsk_setup_xdp_prog(struct xsk_socket *xsk, int *xsks_map_fd)
 	if (IS_ERR(ctx->xdp_prog))
 		return PTR_ERR(ctx->xdp_prog);
 
+	if (ctx->xdp_prog) {
+		int refcnt;
+
+		ctx->refcnt_map_fd = xsk_lookup_refcnt_map(xdp_program__fd(ctx->xdp_prog));
+		if (ctx->refcnt_map_fd == -ENOENT) {
+			pr_warn("AF_XDP refcount map was not found. If you want to use built-in refcount, please recreate your sockets with XDP program that contains int variable in .data section\n");
```

Comment:
I doubt any users are going to understand this message. How about something like: "Another XSK socket was created by a version of libxdp that doesn't support program refcnt, so disabling automatic program removal on unload."